modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/layout/prototype/DefaultSearchLayoutPrototypeCustomizer.java

```diff
@@ -39,37 +40,73 @@
 
 	@Override
 	public void customize(Layout layout) throws Exception {
+		String portletInstanceId = PortletIdCodec.generateInstanceId();
+
 		addBorderlessPortlet(
-			layout, SearchBarPortletKeys.SEARCH_BAR, "column-1");
```

Comment:
Shouldn't you generate an instanceId for each widget separately? @jcampoy

Reply:
it is not a problem to reuse it and also it is better from the performance perspective. what it must be different is the complete portletId in the same page if the porlet is instanceable, but taking into account that we are not repeating these portlets and if the user add  some of them manually to this page it will have a new instanceId, it is not a problem.

 I was thinking about to add something like _autogenerated, but finally I decided to use our API.

Reply:
Your explanation makes sense to me, while performance doesn't seem like it should be considered since this only happens when a new company is generated, I can't think of a reason to need different portletInstanceId suffixes.